### ISSUE TITLE
TDirectory in BCIntegrate

### ISSUE THREAD

**Issue by fredRos** [bat/bat#231]
On my way of fixing #230 I came across [`BCIntegrate::InitializeSATree`](https://github.com/bat/bat/blob/416c7487a8af56ee6ad63927cdf5f0e631f018c5/src/BCIntegrate.cxx#L1198)

Does anyone understand why `gDirectory` is used there? Was is perhaps a poor man's attempt of the guard in #230 but didn't actually modify `gDirectory`?

```c++
    TDirectory* dir = gDirectory;
...
    gDirectory = dir;
```

**Comment by oschulz** [comment]
Looks like it, to me.

**Comment by fredRos** [comment]
fixed by #244
